File: 452-MinimumNumberOfArrowsToBurstBalloons/452-MinimumNumberOfArrowsToBurstBalloons.cpp
// Last updated: 08/05/2026, 09:29:24
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {

        sort(points.begin(), points.end(), [](const auto v,const auto w){return v[1]<w[1];});
        int count=1;
        int n=points.size();
        long current= points[0][1];
        for(int i=1; i<n;i++){
            if(points[i][0]<=current){
                continue;
            }
            current = points[i][1];
            count++;
        }
        return count;
    }
};

File: 452-MinimumNumberOfArrowsToBurstBalloons/452-MinimumNumberOfArrowsToBurstBalloons.py
# Last updated: 08/05/2026, 09:29:28
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points_sorted= sorted(points, key= lambda x:x[1])
        count=0

        print(points_sorted)
        current= float("-inf")

        for x in points_sorted:
            if x[0]<=current:
               continue
               
            else:
                current=x[1]
                count+=1
                

        return count 
